lib/improver/cli/apply_emos_coefficients.py

```diff
@@ -0,0 +1,202 @@
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""Script to apply coefficients for Ensemble Model Output
+Statistics (EMOS), otherwise known as Non-homogeneous Gaussian
+Regression (NGR)."""
+
+import numpy as np
+
+from iris.exceptions import CoordinateNotFoundError
+
+from improver.argparser import ArgParser
+from improver.ensemble_calibration.ensemble_calibration import (
+    ApplyCoefficientsFromEnsembleCalibration)
+from improver.ensemble_copula_coupling.ensemble_copula_coupling import (
+    EnsembleReordering,
+    GeneratePercentilesFromMeanAndVariance,
+    GeneratePercentilesFromProbabilities,
+    GenerateProbabilitiesFromMeanAndVariance,
+    RebadgePercentilesAsRealizations,
+    ResamplePercentiles)
+from improver.utilities.cube_checker import find_percentile_coordinate
+from improver.utilities.load import load_cube
+from improver.utilities.save import save_netcdf
+
+
+def main(argv=None):
+    """Load in arguments for applying coefficients for Ensemble Model Output
+       Statistics (EMOS), otherwise known as Non-homogeneous Gaussian
+       Regression (NGR). The coefficients are applied to the forecast
+       that is supplied, so as to calibrate the forecast. The calibrated
+       forecast is written to a netCDF file.
+    """
+    parser = ArgParser(
+        description='Apply coefficients for Ensemble Model Output '
+                    'Statistics (EMOS), otherwise known as Non-homogeneous '
+                    'Gaussian Regression (NGR)')
+    # Filepaths for the forecast, EMOS coefficients and the output.
+    parser.add_argument('forecast_filepath', metavar='FORECAST_FILEPATH',
+                        help='A path to an input NetCDF file containing the '
+                             'forecast to be calibrated.')
+    parser.add_argument('coefficients_filepath',
+                        metavar='COEFFICIENTS_FILEPATH',
+                        help='A path to an input NetCDF file containing the '
+                             'coefficients used for calibration.')
+    parser.add_argument('output_filepath', metavar='OUTPUT_FILEPATH',
+                        help='The output path for the processed NetCDF')
+    # Optional arguments.
+    parser.add_argument('--num_realizations', metavar='NUMBER_OF_REALIZATIONS',
```

Comment:
What happens with the default now, if there are probabilities or percentiles as input. Does this need to be a compulsory argument now? Or at least saying it will raise an error if not provided

Reply:
I've amended this docstring, as discussed.